test/test_metal.jl

```diff
@@ -39,4 +40,88 @@ Oceananigans.defaults.FloatType = Float32
     @test iteration(simulation) == 3
     @test time(simulation) == 3minutes
 end
+#! format: on
 
+@testset "MetalGPU: ImmersedBoundaryGrid" begin
+    arch = GPU(Metal.MetalBackend())
+    FT = Oceananigans.defaults.FloatType
+
+    Lx, Ly, Lz = FT(5000meters), FT(5000meters), FT(20meters)
+    Nx, Ny, Nz = 16, 16, 8
+
+    underlying_grid = RectilinearGrid(
+        arch;
+        size=(Nx, Ny, Nz),
+        x=(0, Lx),
+        y=(0, Ly),
+        z=(-Lz, 0),
+        topology=(Bounded, Bounded, Bounded),
+        halo=(5, 5, 5),
+    )
+    @test eltype(underlying_grid) == Float32
+
+    @inline function depth(x::FT, y::FT)::FT
+        a² = (FT(0.45) * Lx)^2  # semi-major squared axis
+        b² = (FT(0.45) * Ly)^2  # semi-minor squared axis
+        x₀ = FT(0.5) * Lx
+        y₀ = FT(0.5) * Ly
+        Δx = x - x₀
+        Δy = y - y₀
+        r² = (Δx * Δx) / a² + (Δy * Δy) / b²
+        return r² ≤ FT(1) ? -Lz * (FT(1) - r²) : FT(1)
+    end
+
+    grid = ImmersedBoundaryGrid(
+        underlying_grid,
+        PartialCellBottom(depth);
+        active_cells_map=false,
+    )
+    @test eltype(grid) == Float32
+
+    Qᵀ = FT(1e-4)
+    T_bcs = FieldBoundaryConditions(top=FluxBoundaryCondition(Qᵀ))
+
+    Qᵘ = - FT(1e-4)
+    u_bcs = FieldBoundaryConditions(top=FluxBoundaryCondition(Qᵘ))
+
+    model = HydrostaticFreeSurfaceModel(;
+        grid,
+        coriolis = FPlane(latitude=60),
+        tracers = (:T, :S),
+        buoyancy = SeawaterBuoyancy(),
+        momentum_advection = WENO(),
+        tracer_advection = WENO(),
+        free_surface = SplitExplicitFreeSurface(grid; substeps=30), # default does not work on MetalGPU
+        closure = nothing, # ConvectiveAdjustmentVerticalDiffusivity(convective_κz=1, background_κz=1e-3),
+        boundary_conditions = (u=u_bcs, T=T_bcs),
+    )
+    @test parent(model.velocities.u) isa MtlArray
+    @test parent(model.velocities.v) isa MtlArray
+    @test parent(model.velocities.w) isa MtlArray
+    @test parent(model.tracers.T) isa MtlArray
+    @test parent(model.tracers.S) isa MtlArray
+
+    simulation = Simulation(model, Δt=1, stop_iteration=1)
+    run!(simulation)
+
+    @test iteration(simulation) == 1
+    @test time(simulation) == 1seconds
+end
+
+@testset "MetalGPU: TimeStepWizard" begin
+    arch = GPU(Metal.MetalBackend())
+    FT = Oceananigans.defaults.FloatType
```

Comment:
define FT but not used?

Reply:
`FT` should not be needed in any constructor. AI tools seem to want to pepper `FT` everywhere (we explicitliy write not to do this in AGENTS.md but it does not always seem to get into the AI's head)